bot/exts/valentines/be_my_valentine.py

```diff
@@ -118,56 +109,41 @@ async def send_valentine(
 
-        **This command should be DMed to the bot.**
-
-        syntax : .bemyvalentine secret [user](optional) [p/poem/c/compliment/or you can type your own valentine message]
+        syntax : .bemyvalentine secret [user] [p/poem/c/compliment/or you can type your own valentine message]
         (optional)
 
-        example : .bemyvalentine secret (sends valentine as a poem or a compliment to a random user in DM making you
-        anonymous)
         example : .bemyvalentine secret Iceman#6508 p (sends a poem to Iceman in DM making you anonymous)
         example : .bemyvalentine secret Iceman#6508 Hey I love you, wanna hang around ? (sends the custom message to
         Iceman in DM making you anonymous)
         """
-        if ctx.guild is not None:
-            # This command is only DM specific
-            msg = "You are not supposed to use this command in the server, DM the command to the bot."
-            return await ctx.send(msg)
-
-        if user:
-            if Lovefest.role_id not in [role.id for role in user.roles]:
-                message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
-                return await ctx.send(message)
+        if Lovefest.role_id not in [role.id for role in user.roles]:
+            message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
+            await ctx.message.delete()
+            return await ctx.author.send(message)
 
         if user == ctx.author:
             # Well a user cant valentine himself/herself.
-            return await ctx.send('Come on dude, you cant send a valentine to yourself :expressionless:')
+            await ctx.message.delete()
+            return await ctx.author.send('Come on dude, you cant send a valentine to yourself :expressionless:')
 
-        guild = self.bot.get_guild(id=Client.guild)
         emoji_1, emoji_2 = self.random_emoji()
-        lovefest_role = discord.utils.get(guild.roles, id=Lovefest.role_id)
         valentine, title = self.valentine_check(valentine_type)
 
-        if user is None:
-            author = ctx.author
-            user = self.random_user(author, lovefest_role.members)
-            if user is None:
-                return await ctx.send("There are no users avilable to whome your valentine can be sent.")
-
         embed = discord.Embed(
             title=f'{emoji_1}{title} {user.display_name}{emoji_2}',
             description=f'{valentine} \n **{emoji_2}From anonymous{emoji_1}**',
             color=Colours.pink
         )
         try:
             await user.send(embed=embed)
+            await ctx.message.delete()
         except discord.Forbidden:
-            await ctx.author.send(f"{user} has DMs disabled, so I couldn't send the message. Sorry!")
+            await ctx.send(f"{user} has DMs disabled, so I couldn't send the message. Sorry!")
```

Comment:
What was the reason for changing this from a DM? should we do the same change for the rest of the messages sent via DM in this cog?

Reply:
> Again caused due to intents, the bot was not able to find the user on DMChannel, so i changed it to be passed through the sir-lancebot-commands channel and deletes the their message. Rest it acts similarly, sending the errors/confirmation messages in DMs and sending the valentine message in DMs.

Reply:
`discord.Forbidden` is raised when the bot tries to DM a user, but they have DMs disabled.
In this case, the person the bot was trying to send the valentine to had DMs disabled.

I  this case though, removing the `author` from this changes it so that the bot does contact the author of the command via DM. If this was your intention, I think we should remove the other `ctx.author.send` too, or wrap those in `discord.Forbidden` try/except blocks too.

Reply:
I would create a helper function for this with `try...except` blocks.